app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -100,7 +103,8 @@ def identity_provider
         def extract_resource_restrictions
           @extract_resource_restrictions ||= @extract_resource_restrictions_class.new(
```

Comment:
this is not the way that we use command classes. We initialize them once, run the command they perform and move on. You should not use dependencies in app code, just in UTs